### PR TITLE
wlrctl: init at 0.2.1

### PR DESCRIPTION
**Pull request by puffnfresh**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123848` at 49b31f8b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/eff19275dff71afed671f70d35df8c40">wlrctl</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/wayland/wlrctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchFromSourcehut, cmake, meson, pkg-config, libxkbcommon, wayland, scdoc, ninja }:
+
+stdenv.mkDerivation rec {
+  name = "wlrctl-${version}";
+  version = "0.2.1";
+  src = fetchFromSourcehut {
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wlrctl/default.nix:6]:
```suggestion
  pname = "wlrctl";
  version = "0.2.1";

  src = fetchFromSourcehut {
```

**Comment by SuperSandro2000** [comment]
FTBFS:

```
build flags: -j64 -l64
[19/21] Compiling C object wlrctl.p/keyboard.c.ootos.aon-generated_wlr-foreign-toplevel-management-unstable-v1-protocol.c.o
FAILED: wlrctl.p/keyboard.c.o 
gcc -Iwlrctl.p -I. -I.. -I../include -I/nix/store/vk1inpfffikdxraw16syv2iq71fmrcv2-wayland-1.19.0-dev/include -I/nix/store/qk9j926gbabw8yixii4lbfz74kgrvb3y-libxkbcommon-1.3.0-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -Wno-unused-parameter -Wno-missing-braces '-DWLRCTL_VERSION="0.2.1"' -MD -MQ wlrctl.p/keyboard.c.o -MF wlrctl.p/keyboard.c.o.d -o wlrctl.p/keyboard.c.o -c ../keyboard.c
../keyboard.c: In function 'is_ascii':
../keyboard.c:83:14: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   83 |   if (str[i] < 0) {
      |              ^
cc1: all warnings being treated as errors
```

**Comment by Artturin** [comment]
Result of `nixpkgs-review pr 123848` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wlrctl</li>
  </ul>
</details>

**Comment by Artturin** [comment]
Result of `nixpkgs-review pr 123848` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wlrctl</li>
  </ul>
</details>


tested the example commands and they work
combine the commits https://git-rebase.io/#fixup

**Pull request closed without merging**